src/services/decoders/ClpIrDecoder/index.ts

```diff
@@ -66,7 +73,12 @@ class ClpIrDecoder implements Decoder {
             CLP_IR_STREAM_TYPE.STRUCTURED :
```

Comment:
that we pass the IrNamespaceKeys into the constructor is a bit weird. 
how do you feel about rewriting this factory function to only initialize the module, then we pass the module object into the `ClpIrDecoder` constructor and finish the rest of the initializations.